analyzer/consensus/genesis.go

```diff
@@ -489,24 +491,43 @@ VALUES
 					proposal.Content.Upgrade.Target.RuntimeHostProtocol.String(),
 					proposal.Content.Upgrade.Target.RuntimeCommitteeProtocol.String(),
 					proposal.Content.Upgrade.Epoch,
-					"null",
+					// 1 hardcoded NULL for the proposal.Content.CancelUpgrade.ProposalID field.
+					// 2 hardcoded NULLs for the proposal.Content.ChangeParameters fields.
 					proposal.CreatedAt,
 					proposal.ClosesAt,
 					proposal.InvalidVotes,
 				)
-			} else if proposal.Content.CancelUpgrade != nil {
+			case proposal.Content.CancelUpgrade != nil:
 				query += fmt.Sprintf(
-					"\t(%d, '%s', '%s', %d, NULL, NULL, NULL, NULL, NULL, %d, %d, %d, %d)",
+					"\t(%d, '%s', '%s', %d, NULL, NULL, NULL, NULL, NULL, %d, NULL, NULL, %d, %d, %d)",
 					proposal.ID,
 					proposal.Submitter.String(),
 					proposal.State.String(),
 					proposal.Deposit.ToBigInt(),
 					// 5 hardcoded NULLs for the proposal.Content.Upgrade fields.
 					proposal.Content.CancelUpgrade.ProposalID,
+					// 2 hardcoded NULLs for the proposal.Content.ChangeParameters fields.
 					proposal.CreatedAt,
 					proposal.ClosesAt,
 					proposal.InvalidVotes,
 				)
+			case proposal.Content.ChangeParameters != nil:
+				query += fmt.Sprintf(
+					"\t(%d, '%s', '%s', %d, NULL, NULL, NULL, NULL, NULL, NULL, '%s', decode('%s', 'hex'), %d, %d, %d)",
+					proposal.ID,
+					proposal.Submitter.String(),
+					proposal.State.String(),
+					proposal.Deposit.ToBigInt(),
+					// 5 hardcoded NULLs for the proposal.Content.Upgrade fields.
+					// 1 hardocded NULL for the proposal.Content.CancelUpgrade.ProposalID field.
+					proposal.Content.ChangeParameters.Module,
+					hex.EncodeToString(proposal.Content.ChangeParameters.Changes),
```

Comment:
:facepalm: one day we'll use a sql library in genesis processing too ... it's just not justifiable to spend a day doing that busy work.